src/data/resolvers/queries/deals.ts

```diff
@@ -48,17 +48,26 @@ const dealQueries = {
   /**
    * Get last board
    */
+
   dealBoardGetLast() {
     return DealBoards.findOne().sort({ createdAt: -1 });
   },
 
   /**
    * Deal Pipelines list
    */
-  dealPipelines(_root, { boardId }: { boardId: string }) {
-    return DealPipelines.find({ boardId }).sort({ order: 1, createdAt: -1 });
-  },
+  dealPipelines(_root, { boardId }: { boardId: string }, { user }: { user: IUserDocument }) {
+    if (user.isOwner) {
+      return DealPipelines.find({ boardId }).sort({ order: 1, createdAt: -1 });
+    }
 
+    return DealPipelines.find({
+      $and: [
+        { boardId },
+        { $or: [{ type: 'public' }, { type: { $exists: false } }, { memberIds: user._id }, { userId: user._id }] },
+      ],
+    }).sort({ order: 1, createdAt: -1 });
+  },

```

Comment:
Separate into two different functions and refactor this function. And add migrate.